src/coreComponents/physicsSolvers/fluidFlow/IsothermalCompositionalMultiphaseFVMKernels.hpp

```diff
@@ -717,37 +721,111 @@ class FaceBasedAssemblyKernel : public FaceBasedAssemblyKernelBase
           localIndex const esr_up = sesri[k_up];
           localIndex const ei_up  = sei[k_up];
 
-          real64 const mobility = m_phaseMob[er_up][esr_up][ei_up][ip];
-
-          // skip the phase flux if phase not present or immobile upstream
-          if( LvArray::math::abs( mobility ) < 1e-20 ) // TODO better constant
+          // C1PPU
+          if( m_useC1PPU > 0 && m_epsC1PPU > 0 )
           {
-            continue;
-          }
+            GEOS_ASSERT( numFluxSupportPoints == 2 );
+
+            real64 const mobility_i = m_phaseMob[seri[0]][sesri[0]][sei[0]][ip];
+            real64 const mobility_j = m_phaseMob[seri[1]][sesri[1]][sei[1]][ip];
```

Comment:
@NirvanaGodJia can answer better, up to my understanding it should not matter, see also the paper equation 66

Reply:
As discussed with Pavel and Jacques: one thing to consider is the fact that indices are going to change when we change the MPI partitions (changing the number of ranks, or simply changing the shape of the partitions). We want to make sure that the results produced by the flux kernels are independent of the change in indices. We can discuss it later again if needed.

Reply:
> 

Thanks for the comment. Should not relate to upwind and downwind because the motivation of smooth flux is avoiding switches based on upwinding direction. If the code still needs to decide based on a switching criterion, then there will still be discontinuity issue.